datastream-server/src/test/java/com/linkedin/datastream/server/zk/TestZkAdapter.java

```diff
@@ -1059,6 +1061,59 @@ public void testUpdateAssignmentDeleteUnusedTasks() {
     adapter.disconnect();
   }
 
+  @Test
+  public void testUpdateAllAssignmentAndIssueTokens() throws Exception {
```

Comment:
Can we add a test for failure cases too? Like in your code https://github.com/linkedin/brooklin/pull/919/files#diff-c0704e6fe31835d24d153416a6d6efb590a4458c334d83326cd61e64c6ce9e8eR759 you have filter for stopping DS tasks only. Can we add test where we have mix of task(stopping and non-stopping) and make sure the code takes only stopping tasks?

Reply:
That's what my current test is doing. It's creating 2 datastreams and only one of them is stopping. Please refer to the assertions I have there.